### PR TITLE
#4638 fix nullpointer exception in user detail 

### LINKED ISSUE #4638: Nullpointer exception in user detail
When user detail is opened. Nullpointer exception is thrown.

Problem is that user has ldap group but no ldap server is assigned.

`
java.lang.NullPointerException: null
	at org.kitodo.production.services.data.LdapServerService.createNewUser(LdapServerService.java:165) ~[classes/:3.4.0-SNAPSHOT]
	at org.kitodo.production.forms.UserForm.writeUserAtLdapServer(UserForm.java:489) [classes/:3.4.0-SNAPSHOT]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.apache.el.parser.AstValue.invoke(AstValue.java:246) [jasper-el.jar:8.5.70]
	at org.apache.el.MethodExpressionImpl.invoke(MethodExpressionImpl.java:266) [jasper-el.jar:8.5.70]
	at org.jboss.weld.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
	at org.jboss.weld.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
	at org.jboss.weld.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
	at org.jboss.weld.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
	at org.apache.myfaces.view.facelets.el.ContextAwareTagMethodExpression.invoke(ContextAwareTagMethodExpression.java:96) [myfaces-impl-2.3.4.jar:2.3.4]
	at org.apache.myfaces.application.ActionListenerImpl.processAction(ActionListenerImpl.java:74) [myfaces-impl-2.3.4.jar:2.3.4]
	at org.springframework.faces.webflow.FlowActionListener.processAction(FlowActionListener.java:71) [spring-faces-2.5.1.RELEASE.jar:2.5.1.RELEASE]
	at org.springframework.faces.model.SelectionTrackingActionListener.processAction(SelectionTrackingActionListener.java:64) [spring-faces-2.5.1.RELEASE.jar:2.5.1.RELEASE]
	at javax.faces.component.UICommand.broadcast(UICommand.java:120) [myfaces-api-2.3.4.jar:2.3.4]
	at javax.faces.component.UIViewRoot._broadcastAll(UIViewRoot.java:1255) [myfaces-api-2.3.4.jar:2.3.4]
	at javax.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:420) [myfaces-api-2.3.4.jar:2.3.4]
	at javax.faces.component.UIViewRoot._process(UIViewRoot.java:1741) [myfaces-api-2.3.4.jar:2.3.4]
	at javax.faces.component.UIViewRoot.processDecodes(UIViewRoot.java:942) [myfaces-api-2.3.4.jar:2.3.4]
	at org.apache.myfaces.lifecycle.ApplyRequestValuesExecutor.execute(ApplyRequestValuesExecutor.java:42) [myfaces-impl-2.3.4.jar:2.3.4]
	at org.apache.myfaces.lifecycle.LifecycleImpl.executePhase(LifecycleImpl.java:195) [myfaces-impl-2.3.4.jar:2.3.4]
	at org.apache.myfaces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:142) [myfaces-impl-2.3.4.jar:2.3.4]
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:204) [myfaces-api-2.3.4.jar:2.3.4]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) [tomcat-websocket.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.kitodo.production.servletfilter.EncodingFilter.doFilter(EncodingFilter.java:69) [classes/:3.4.0-SNAPSHOT]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:317) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.kitodo.production.security.SecurityObjectAccessFilter.doFilter(SecurityObjectAccessFilter.java:86) [classes/:3.4.0-SNAPSHOT]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:114) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:170) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.session.ConcurrentSessionFilter.doFilter(ConcurrentSessionFilter.java:155) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:200) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:66) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.23.RELEASE.jar:4.3.23.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.23.RELEASE.jar:4.3.23.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:214) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:177) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:347) [spring-web-4.3.23.RELEASE.jar:4.3.23.RELEASE]
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:263) [spring-web-4.3.23.RELEASE.jar:4.3.23.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.logging.log4j.web.Log4jServletFilter.doFilter(Log4jServletFilter.java:71) [log4j-web-2.13.2.jar:2.13.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:196) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) [catalina.jar:8.5.70]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:544) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135) [catalina.jar:8.5.70]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) [catalina.jar:8.5.70]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:698) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) [catalina.jar:8.5.70]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:364) [catalina.jar:8.5.70]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:624) [tomcat-coyote.jar:8.5.70]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) [tomcat-coyote.jar:8.5.70]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:831) [tomcat-coyote.jar:8.5.70]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1650) [tomcat-coyote.jar:8.5.70]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:8.5.70]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) [tomcat-util.jar:8.5.70]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) [tomcat-util.jar:8.5.70]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.5.70]
	at java.lang.Thread.run(Thread.java:829) [?:?]
`

### PR DESCRIPTION
**Pull request by markusweigelt**
Fixes #4638

When user detail/edit is opened we check if ldap server is not null.

If ldap server is null a helper message will be shown.

### CODE REVIEW

**Comment by solth** [comment]
If you link your pull request to the issue mentioned in the title a la "Fixes #4638" in the description, the issue will be closed automatically once the pull request gets merged. (see https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue for details)

File: Kitodo/src/main/java/org/kitodo/production/services/data/LdapServerService.java [hunk 1]
```diff
@@ -162,6 +162,11 @@ public void createNewUser(User user, String password)
             return;
         }
 
+        if ( user.getLdapGroup().getLdapServer() == null) {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/services/data/LdapServerService.java:165]: We use `Objects.isNull()` for nullChecks (see 5 lines above). It's a convention, and I would like to keep it consistent.
If this is a good idea can be discussed here: #4087

File: Kitodo/src/main/java/org/kitodo/production/services/data/LdapServerService.java [hunk 2]
```diff
@@ -162,6 +162,11 @@ public void createNewUser(User user, String password)
...
+            Helper.setMessage(Helper.getTranslation("noLdapServerAssignedToLdapGroup"));
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/services/data/LdapServerService.java:166]: My first intention was, that `Helper.setErrorMessage` should be used here, but it is also a regular message the lines above. 🤦 
Please change to errorMessage in both places, or open an issue and fix it in a separat PR (which would be the better way 😏  )

**Review by Kathrin-Huber — CHANGES_REQUESTED** [review]

**Comment by markusweigelt** [comment]
Changed null check using null check convention. To change log level of error messages issue #4645 is created.

**Review by markusweigelt — COMMENTED** [review]
Requested changes are resolved.

**Review by Kathrin-Huber — APPROVED** [review]

**Pull request merged**